src/c++/perf_analyzer/genai-perf/genai_perf/main.py

```diff
@@ -137,15 +137,19 @@ def finalize(profile_export_file: Path):
 # to assert correct errors and messages.
 def run():
     try:
+        # TMA-1900: refactor CLI handler
         init_logging()
         args, extra_args = parser.parse_args()
-        create_artifacts_dirs(args.generate_plots)
-        tokenizer = get_tokenizer(args.tokenizer)
-        generate_inputs(args, tokenizer)
-        args.func(args, extra_args)
-        data_parser = calculate_metrics(args, tokenizer)
-        report_output(data_parser, args)
-        finalize(args.profile_export_file)
+        if args.subcommand == "compare":
```

Comment:
I think an if/elif/else is fine here. I would prefer we refactor the calls inside the ifs to something like 
`if args.subcommand == "compare":
   compare_workflow()
 else:
    profile_workflow()
`
I want to make the main loop more concise with this addition and we can move the details to a method call.

thoughts?

Reply:
Yes, I agree and that's why I put a refactor ticket on top.

Reply:
Does this refactor have a larger scope than just this piece?

Reply:
I think it will require more changes to the existing code. More cleaner approach will look something like:
```python
def run():
    try:
        init_logging()
        args, extra_args = parser.parse_args()
        args.func(args, extra_args)
    except Exception as e:
        raise GenAIPerfException(e)
```
where `args.func` runs either the profile (and launches PA) or the compare workflow. However, in order to do this, we need move around the codes and that might break other things. This was the most naive but simple way to add subcommand functionality on top of our existing workflow.

Reply:
Just trying to understand, is there a possibility to have more subcommands?

Reply:
right now, i dont think so.
but we are discussing moving the default workflow to a subcommand as well.

Reply:
@nv-hwoo i think we can close this one for now. You tagged it as a refactor and if you think its more work than a simple change, lets delay to the other ticket. Thanks for explaining it to me.